src/GitHubExtension/DeveloperId/DeveloperIdProvider.cs

```diff
@@ -255,6 +287,20 @@ private DeveloperId CreateOrUpdateDeveloperId(OAuthRequest oauthRequest)
                 Log.Logger()?.ReportError($"LoggedIn event signaling failed: {error}");
             }
         }
+    }
+
+    private DeveloperId CreateOrUpdateDeveloperIdFromOauthRequest(OAuthRequest oauthRequest)
+    {
+        // Query necessary data and populate Developer Id.
+        var newDeveloperId = oauthRequest.RetrieveDeveloperId();
+        var accessToken = oauthRequest.AccessToken;
+        if (accessToken is null)
+        {
+            Log.Logger()?.ReportError($"Invalid AccessToken");
+            throw new InvalidOperationException();

```

Comment:
I noticed on line 122 you added text to the exception, is that something you want to do here as well?  Or we're you ok with just the invalid operation exception with no text?

Reply:
This is fine with no text. These exceptions don't reach DevHome, and will only be shown as a Generic message on the LoginUI for now. The logs will show detailed code flow for us to debug though.